src/net/mod.rs

```diff
@@ -209,29 +209,17 @@ pub fn get_device_prefix(
     prefix
 }
 
-/// Determine the broker IP address
-///
-/// # Note
-/// If the broker IP is unspecified or unparseable, the default IP is returned.
+/// Parse a string containing an IPv4 address.
+/// This is a convenience wrapper around the `no_std_net` parser.
 ///
 /// # Args
-/// * `input` - The optionally-specified command-line broker IP address as a string.
+/// * `input` - The IP address as a string.
 ///
 /// # Returns
-/// The broker IP address.
-pub fn parse_or_default_broker(input: Option<&str>) -> IpAddr {
-    input.and_then(|data| {
-             data.parse::<minimq::embedded_nal::IpAddr>().map_or_else(
-                 |err| {
-                     log::error!(
-                         "{:?}: Failed to parse broker IP ({:?}) - Falling back to default",
-                         err,
-                         data
-                     );
-                     None
-                 },
-                 Some,
-             )
-         })
-         .unwrap_or_else(|| DEFAULT_MQTT_BROKER.into())
+/// The IP address.
+///
+/// # Panic
+/// This panics if parsing fails.
+pub fn parse_ipv4(input: &str) -> IpAddr {
+    input.parse::<minimq::embedded_nal::IpAddr>().unwrap()
```

Comment:
1. I don't think you need to tell `parse()` the type here because it should be able to infer it via the function return type - or was it giving you errors otherwise?
2. This implementation will panic if the BROKER command-line argument was invalid - I was debating that behavior originally, but opted for defaulting back to the default broker. I can understand why just straight panicking may be more desirable, but I just want to make sure this is intended

Reply:
Right. I just hadn't noticed I can remove the type.
I'm for panicing here. If the user specifies a broker IP at compile time that means the default broker will not work for them. If the user specifies an invalid IP it doesn't make much sense to proceed with a default broker because we know it won't work. And since the intention was to get a working broker connection, we should rather fail/panic than try something that we know won't work.